repos/system_upgrade/el7toel8/actors/migratentp/libraries/ntp2chrony.py

```diff
@@ -0,0 +1,676 @@
+                options[words[0]] = float(words[1])
+                words = words[2:]
+            else:
+                return False
+
+        self.tos_options.update(options)
+
+        return True
+
+    def parse_includefile(self, words):
+        path = self.root_dir + words[0]
+        if not os.path.isfile(path):
+            return False
+
+        self.parse_ntp_conf(path)
+        return True
+
+    def parse_keys(self, words):
+        keyfile = words[0]
+        path = self.root_dir + keyfile
+        if not os.path.isfile(path):
+            logging.info("Missing %s", path)
+            return False
+
+        with open(path, encoding=self.file_encoding) as f:
+            logging.info("Reading %s", path)
+            keys = []
+            for line in f:
+                words = line.split()
+                if len(words) < 3 or not words[0].isdigit():
+                    continue
+                keys.append((int(words[0]), words[1], words[2]))
+
+            self.keyfile = keyfile
+            self.keys = keys
+
+        return True
+
+    def parse_trustedkey(self, words):
+        key_ranges = []
+        for word in words:
+            if word.isdigit():
+                key_ranges.append((int(word), int(word)))
+            elif re.match("^[0-9]+-[0-9]+$", word):
+                first, last = word.split("-")
+                key_ranges.append((int(first), int(last)))
+            else:
+                return False
+
+        self.trusted_keys = key_ranges
+        return True
+
+    def write_chrony_configuration(self, chrony_conf_path, chrony_keys_path,
+                                   dry_run=False, backup=False):
+        chrony_conf = self.get_chrony_conf(chrony_keys_path)
+        logging.debug("Generated %s:\n%s", chrony_conf_path, chrony_conf)
+
+        if not dry_run:
+            self.write_file(chrony_conf_path, 0o644, chrony_conf, backup)
+
+        chrony_keys = self.get_chrony_keys()
+        if chrony_keys:
+            logging.debug("Generated %s:\n%s", chrony_keys_path, chrony_keys)
+
+        if not dry_run:
+            self.write_file(chrony_keys_path, 0o640, chrony_keys, backup)
+
+    def get_processed_time_sources(self):
+        # Convert {0,1,2,3}.*pool.ntp.org servers to 2.*pool.ntp.org pools
+
+        pools = {}
+        for source in self.time_sources:
+            if source["type"] != "server":
+                continue
+            if not source["address"].endswith(".pool.ntp.org"):
+                continue
+            m = re.match("^([0123])(\\.\\w+)?\\.pool\\.ntp\\.org$", source["address"])
+            if m is None:
+                continue
+            number = m.group(1)
+            zone = m.group(2)
+            if zone not in pools:
+                pools[zone] = []
+            pools[zone].append((number, source))
+
+        remove_servers = set()
+        convert_servers = set()
+        for zone, pool in pools.items():
+            pool.sort()
```

Comment:
Do you mean lexicographic check here, or are you missing int conversion for pool[0]?